oci/spec_opts_test.go

```diff
@@ -464,6 +464,36 @@ func TestWithTTYSize(t *testing.T) {
 
 }
 
+func ContainsLinuxIDMapping(a []specs.LinuxIDMapping, x specs.LinuxIDMapping) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+func TestWithUserNamespace(t *testing.T){
+	t.Parallel()
+	s := Spec{}
+	opts := []SpecOpts{
+		WithUserNamespace(1, 2,20000),
+	}
+	for _, opt := range opts {
+		if err := opt(nil, nil, nil, &s); err != nil {
+			t.Fatal(err)
+		}
+	}
+	testMapping := specs.LinuxIDMapping{
+		ContainerID: 1,
+		HostID:      2,
+		Size:        20000,
+	}
+	if !ContainsLinuxIDMapping(s.Linux.UIDMappings,testMapping) || !ContainsLinuxIDMapping(s.Linux.GIDMappings,testMapping) {
```

Comment:
I'm not sure you need the function `ContainsLinuxIDMapping` as there should only be one (the one you set up in `SpecOpts`), so a more correct test would be to:
 1. verify there is only one entry in the array (`len(s.Linux.UIDMappings) == 1`) and,
 2. that the one entry is the one you specified: `s.Linux.UIDMappings[0] == testMapping`

Of course you need to check both UIDMappings and GIDMappings have this single entry.